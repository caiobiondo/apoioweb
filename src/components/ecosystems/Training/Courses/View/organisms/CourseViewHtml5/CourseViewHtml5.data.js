import gql from 'graphql-tag';

export const CourseViewHtml5Query = gql`
  query CourseViewHtml5Query(
    $sellerId: Int!
    $courseId: Int!
    $ciclo: Int
    $setor: Int
    $gerenciaMercado: Int
    $grupo: Int
    $papelDaConsultora: Int
    $canal: Int
    $gerenciaDeVendas: Int
    $regiao: Int
  ) {
    courseHtml5(
      sellerId: $sellerId
      courseId: $courseId
      ciclo: $ciclo
      setor: $setor
      gerenciaMercado: $gerenciaMercado
      grupo: $grupo
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      gerenciaDeVendas: $gerenciaDeVendas
      regiao: $regiao
    ) {
      id
      courseContent {
        video
        html5
        web
        videoEmbed
        html5Embed
      }
      durationInSeconds
      description
      generalRating
      isfavorite
      myRating
      ratedByYou
      relatedCourses {
        id
        title
        thumbnail
        type
        url
      }
      status
      stoppedAt
      thumbnail
      title
      type
      views
    }
  }
`;

export const CourseViewHtml5QueryOptions = {
  options(props) {
    return {
      variables: {
        sellerId: props.user.codigo,
        courseId: props.courseId,
        ciclo:
          props.user.estrutura.ciclo &&
          props.user.estrutura.ciclo[0] &&
          props.user.estrutura.ciclo[0].numero,
        setor: props.user.estrutura.setor.codigo,
        gerenciaMercado: props.user.estrutura.gerenciaMercado.codigo,
        grupo: props.user.estrutura.codigo,
        papelDaConsultora: props.user.cdPapelAtivo,
        canal: props.user.cdCanalCaptacao,
        gerenciaDeVendas: props.user.estrutura.gerenciaVenda.codigo,
        regiao: props.user.estrutura.regiaoEstrategica.codigo,
      },
      forceFetch: true,
    };
  },

  props({ data }) {
    return {
      data,
      loading: data.loading,
      course: data.courseHtml5,
      refetch: data.refetch,
    };
  },
};
