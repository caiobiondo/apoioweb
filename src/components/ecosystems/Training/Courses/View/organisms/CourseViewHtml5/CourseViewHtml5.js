import React, { Component } from 'react';
import { graphql, compose, withApollo } from 'react-apollo';
import { withRouter } from 'react-router-dom';
import { injectIntl, FormattedMessage } from 'react-intl';
import gql from 'graphql-tag';

import { CourseViewHtml5Query, CourseViewHtml5QueryOptions } from './CourseViewHtml5.data';
import { TrainingCourseUpdateMutation } from 'components/ecosystems/Training/data/TrainingCourseUpdate.data';
import {
  TrainingCoursesQuery,
  TrainingCoursesQueryOptions,
} from 'components/ecosystems/Training/data/TrainingCourses.data';

import {
  Main,
  CourseViewFeedbackModalTitle,
  CourseViewFeedbackModalAction,
  TrainingCourseThumbnail,
  TrainingCourseThumbnailWrapper,
  TrainingCourseThumbnailDescriptionWrapper,
  TrainingCourseActions,
  TrainingCourseActionButtonMobile,
  TrainingCourseActionButtonWrapper,
  Html5Wrapper,
  IconWrapper,
  TrainingCourseUnavailableTitle,
} from './CourseViewHtml5.styles';

import CourseViewHeader from 'components/ecosystems/Training/Courses/View/molecules/CourseViewHeader';
import CourseDescription from 'components/ecosystems/Training/Courses/View/molecules/CourseDescription';
import RelatedCourses from 'components/ecosystems/Training/Courses/View/molecules/RelatedCourses';
import CourseRating from 'components/ecosystems/Training/Courses/View/molecules/CourseRating';
import CourseEvaluation from 'components/ecosystems/Training/Courses/View/molecules/CourseEvaluation';
import Dialog from 'material-ui/Dialog';
import { translate } from 'locale';

import { Loading, FlatButton, Icon } from 'natura-ui';

export class CourseViewHtml5 extends Component {
  state = {
    feedbackModalOpened: false,
    feedbackModalTitle: '',
    showEvaluation: false,
    course: {},
  };

  componentDidMount() {
    const { course } = this.props;
    if (course) this.setState({ course });
  }

  componentWillReceiveProps({ loading, course }) {
    this.notifyLoadFinish(loading, course);

    if (!this.props.course && course) {
      this.setState({ course });
    }
  }

  componentDidUpdate() {
    if (
      !this.props.course ||
      !this.refs.html5Iframe ||
      this.refs.html5Iframe.contentDocument.body.innerHTML
    ) {
      return;
    }

    this.refs.html5Iframe.contentDocument.open();
    this.refs.html5Iframe.contentDocument.write(this.props.course.courseContent.html5Embed);
    this.refs.html5Iframe.contentDocument.close();
  }

  notifyLoadFinish = (loading, course) => {
    if (!loading && this.props.onLoadFinished) {
      this.props.onLoadFinished(this.isEmpty(loading, course), this.isLoading(loading, course));
    }
  };

  isLoading = (loading, course) => loading && !course;

  isEmpty = (loading, course) => !loading && !course.id;

  myListIconName = () => {
    return this.state.course.isfavorite === 'true' ? 'ico_minus' : 'ico_plus';
  };

  valueToUpdateMyList = () => {
    return this.state.course.isfavorite === 'true' ? 'unfavorite' : 'favorite';
  };

  handleTrainingError = () => {
    const { formatMessage } = this.props.intl;

    this.setState({
      feedbackModalOpened: true,
      feedbackModalTitle: formatMessage({ id: 'trainingStatusUpdateError' }),
    });
  };

  handleMyListError = () => {
    this.handleDefaultMyList('trainingAddCourseError', 'trainingRemoveCourseError');
  };

  handleNotUpdateMyList = () => {
    this.handleDefaultMyList('trainingAddCourseFailure', 'trainingRemoveCourseFailure');
  };

  handleUpdateSuccessMyList = () => {
    this.handleDefaultMyList('trainingAddCourseSuccess', 'trainingRemoveCourseSuccess');
    const isfavorite = this.state.course.isfavorite === 'true' ? 'false' : 'true';
    this.setState({ course: { ...this.state.course, isfavorite } }, this.updateCachedList);
  };

  handleDefaultMyList = (addMsg, removeMsg) => {
    const { formatMessage } = this.props.intl;
    const message =
      this.valueToUpdateMyList() === 'favorite'
        ? formatMessage({ id: addMsg })
        : formatMessage({ id: removeMsg });

    this.setState({
      feedbackModalOpened: true,
      feedbackModalTitle: message,
    });
  };

  handleTrainingClick = action => event => {
    const { course } = this.props;

    this.props
      .mutate({
        variables: {
          input: { action },
          sellerId: this.props.user.codigo,
          courseId: course.id,
        },
      })
      .then(response => {
        if (action === 'initialized') {
          this.setState(
            { course: { ...this.state.course, status: 'started' } },
            this.updateCachedList,
          );

          if (course.type === 'WEB') window.open(course.courseContent.web, '_blank');
          if (course.type === 'VIDEO') {
            this.props.history.push(`/training/courses/${course.id}/video`);
          }
        }

        if (action === 'terminated') {
          this.setState(
            {
              showEvaluation: true,
              course: { ...this.state.course, status: 'finished' },
            },
            this.updateCachedList,
          );
        }

        return;
      })
      .catch(err => {
        console.log('err', err);

        this.handleTrainingError();
      });
  };

  updateCachedList = () => {
    const { course } = this.state;
    const { client } = this.props;

    client.writeFragment({
      id: course.id,
      fragment: gql`
        fragment myCourse on Course {
          isfavorite
          status
          __typename
        }
      `,
      data: {
        isfavorite: course.isfavorite,
        status: course.status,
        __typename: 'Course',
      },
    });

    let startedCourses = null;
    try {
      startedCourses = client.readQuery({
        query: TrainingCoursesQuery,
        variables: TrainingCoursesQueryOptions.options({
          ...this.props,
          status: 'started',
        }).variables,
      });
    } catch (e) {
      // could not find cache
    }

    if (startedCourses) {
      if (this.state.course.status === 'started' && this.props.course.status === 'pending') {
        startedCourses.courses.items.push(this.state.course);
      }

      if (this.state.course.status !== 'started' && this.props.course.status === 'started') {
        startedCourses.courses.items = startedCourses.courses.items.filter(
          item => item.id !== this.state.course.id,
        );
      }

      client.writeQuery({
        query: TrainingCoursesQuery,
        variables: TrainingCoursesQueryOptions.options({
          ...this.props,
          status: 'started',
        }).variables,
        data: startedCourses,
      });
    }

    let favoritedCourses = null;
    try {
      favoritedCourses = client.readQuery({
        query: TrainingCoursesQuery,
        variables: TrainingCoursesQueryOptions.options({
          ...this.props,
          favorite: true,
        }).variables,
      });
    } catch (e) {
      // could not find cache
    }

    if (favoritedCourses) {
      if (this.state.course.isfavorite === 'true' && this.props.course.isfavorite === 'false') {
        favoritedCourses.courses.items.push(this.state.course);
      }

      if (this.state.course.isfavorite === 'false' && this.props.course.isfavorite === 'true') {
        favoritedCourses.courses.items = favoritedCourses.courses.items.filter(item => {
          return item.id !== this.state.course.id;
        });
      }

      client.writeQuery({
        query: TrainingCoursesQuery,
        variables: TrainingCoursesQueryOptions.options({
          ...this.props,
          favorite: true,
        }).variables,
        data: favoritedCourses,
      });
    }
  };

  handleMyListClick = (event, child) => {
    const { course } = this.props;
    this.props
      .mutate({
        variables: {
          input: { action: this.valueToUpdateMyList() },
          sellerId: this.props.user.codigo,
          courseId: course.id,
        },
      })
      .then(response => {
        if (response.error) {
          this.handleMyListError();
          return;
        }

        if (!response.data.updateCourse.status) {
          this.handleNotUpdateMyList();
          return;
        }

        this.handleUpdateSuccessMyList();
        return;
      })
      .catch(err => {
        console.log('err', err);

        this.handleMyListError();
      });
  };

  handleClose = () => {
    this.setState({ feedbackModalOpened: false });
  };

  renderFeedbackModal = () => {
    const { feedbackModalTitle } = this.state;
    const actions = [
      <FlatButton
        label={<FormattedMessage id="ok" />}
        primary={true}
        labelStyle={CourseViewFeedbackModalAction}
        onClick={this.handleClose}
      />,
    ];

    return (
      <Dialog
        key="feedbackModal"
        title={feedbackModalTitle}
        actions={actions}
        modal={false}
        open={this.state.feedbackModalOpened}
        titleStyle={CourseViewFeedbackModalTitle}
        onRequestClose={this.handleClose}
      />
    );
  };

  renderActionButtons = (buttonStyle, course) => {
    const buttons = [];
    if (course.status === 'pending') {
      buttons.push(
        <TrainingCourseActionButtonWrapper key="start">
          <FlatButton
            {...buttonStyle}
            label={translate('startTraining')}
            icon={<Icon file="ico_play_circle" />}
            onClick={this.handleTrainingClick('initialized')}
          />
        </TrainingCourseActionButtonWrapper>,
      );
    }

    if (course.status === 'started' || course.status === 'paused') {
      buttons.push(
        <TrainingCourseActionButtonWrapper key="finish">
          <FlatButton
            {...buttonStyle}
            label={translate('finishTraining')}
            icon={<Icon file="ico_play_circle" />}
            onClick={this.handleTrainingClick('terminated')}
          />
        </TrainingCourseActionButtonWrapper>,
      );
    }

    return [
      ...buttons,
      <TrainingCourseActionButtonWrapper key="list">
        <FlatButton
          {...buttonStyle}
          label={translate('myList')}
          icon={<Icon file={this.myListIconName()} />}
          onClick={this.handleMyListClick}
        />
      </TrainingCourseActionButtonWrapper>,
    ];
  };

  canEvaluate = () => {
    const { course } = this.props;
    return this.state.showEvaluation && course.ratedByYou !== 'true';
  };

  isCourseAvailable = () => {
    const { course } = this.props;

    return !/.*\.zip$/i.test(course.courseContent.html5);
  };

  render() {
    const { course, loading } = this.props;

    if (!course && loading) {
      return <Loading background="transparent" />;
    }

    if (!course.id) return null;

    return (
      <Main>
        <CourseViewHeader />
        <TrainingCourseThumbnailWrapper>
          {!this.isCourseAvailable() && (
            <TrainingCourseThumbnail imageUrl={course.thumbnail}>
              <TrainingCourseThumbnailDescriptionWrapper>
                <IconWrapper>
                  <Icon file="ico_graduate_cap" />
                </IconWrapper>
                <TrainingCourseUnavailableTitle>
                  {translate('trainingCourseUnavailable')}
                </TrainingCourseUnavailableTitle>
              </TrainingCourseThumbnailDescriptionWrapper>
            </TrainingCourseThumbnail>
          )}
          {this.isCourseAvailable() && (
            <Html5Wrapper>
              <iframe
                ref="html5Iframe"
                title={course.title}
                src="about:blank"
                allowFullScreen
                frameBorder="0"
              />
            </Html5Wrapper>
          )}
        </TrainingCourseThumbnailWrapper>
        <CourseDescription course={course} />
        {this.isCourseAvailable() && (
          <TrainingCourseActions>
            {this.renderActionButtons(TrainingCourseActionButtonMobile, course)}
          </TrainingCourseActions>
        )}
        <CourseRating course={course} />
        <RelatedCourses courses={course.relatedCourses} />
        {this.isCourseAvailable() && this.renderFeedbackModal()}
        {this.isCourseAvailable() &&
          this.canEvaluate() && (
            <CourseEvaluation courseId={course.id} sellerId={this.props.user.codigo} />
          )}
      </Main>
    );
  }
}

export const CourseViewHtml5WithIntl = injectIntl(CourseViewHtml5);
export const CourseViewHtml5WithRouter = withRouter(CourseViewHtml5WithIntl);
export const CourseViewHtml5WithApollo = withApollo(CourseViewHtml5WithRouter);

export default compose(
  graphql(CourseViewHtml5Query, CourseViewHtml5QueryOptions),
  graphql(TrainingCourseUpdateMutation),
)(CourseViewHtml5WithApollo);
