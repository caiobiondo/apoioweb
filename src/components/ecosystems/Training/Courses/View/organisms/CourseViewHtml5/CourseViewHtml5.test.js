import React from 'react';
import { shallow } from 'enzyme';
import { CourseViewHtml5 } from './CourseViewHtml5';

const setup = propOverrides => {
  const intl = {
    formatMessage: value => `formatedMessage ${value}`,
  };
  const client = {
    writeFragment: options => `writeFragment ${options}`,
    readQuery: options => `readQuery ${options}`,
    writeQuery: options => `writeQuery ${options}`,
  };
  global.dataLayer = {
    push: jest.fn(),
  };
  global.open = jest.fn();

  const props = Object.assign(
    {
      user: {
        codigo: 1234,
      },
      course: {
        id: 1,
        title: 'new course',
        description: 'new course description',
        durationInSeconds: 1234,
        stoppedAt: 12,
        views: 12,
        dateUpload: '2017-04-20T00:00:00.000Z',
        type: 'VIDEO',
        status: 'started',
        isfavorite: true,
        relatedCourses: [],
        ratedByYou: 'false',
        courseContent: {
          html5: 'file.html',
          html5Embed: '<html></html>',
        },
      },
      history: { push: jest.fn() },
      mutate: jest.fn().mockReturnValue(
        Promise.resolve({
          data: {
            updateCourse: {
              status: true,
              message: 'success',
            },
          },
        }),
      ),
      onLoadFinished: jest.fn(),
      refetch: jest.fn(),
      intl,
      client,
    },
    propOverrides,
  );

  const result = shallow(<CourseViewHtml5 {...props} />);

  return {
    props,
    result,
  };
};

describe('TrainingCourseViewHtml5', () => {
  it('renders correctly when loading', () => {
    // given
    // when
    const { result } = setup({ loading: true, course: null });

    // then
    expect(result).toMatchSnapshot();
  });

  it('renders correctly when the course is not defined', () => {
    // given
    // when
    const { result } = setup({ loading: false, course: {} });

    // then
    expect(result).toMatchSnapshot();
  });

  it('renders correctly when the course is defined', () => {
    // given
    // when
    const { result } = setup({ loading: false });

    // then
    expect(result).toMatchSnapshot();
  });

  it('should notify onLoadFinished callback when not loading', () => {
    // given
    // when
    const { result, props } = setup({
      loading: false,
    });
    result.instance().componentWillReceiveProps({ loading: false, course: props.course });

    // then
    expect(props.onLoadFinished).toBeCalledWith(false, false);
  });

  describe('when handleMyListClick', () => {
    it('correctly call mutation', () => {
      // given

      // when
      const { result, props } = setup({ loading: false });
      result.instance().handleMyListClick();

      // then
      expect(props.mutate).toBeCalled();
    });
  });

  describe('when handleTrainingClick', () => {
    it('correctly call mutation', () => {
      // given

      // when
      const { result, props } = setup({ loading: false });
      result.instance().handleTrainingClick('initialized')();

      // then
      expect(props.mutate).toBeCalled();
    });

    describe('when finishing a training course', () => {
      it('asks for a course evaluation', async () => {
        // given
        const course = {
          id: 1,
          title: 'new course',
          description: 'new course description',
          durationInSeconds: 1234,
          stoppedAt: 12,
          views: 12,
          dateUpload: '2017-04-20T00:00:00.000Z',
          type: 'VIDEO',
          status: 'pending',
          isfavorite: true,
          relatedCourses: [],
          ratedByYou: 'false',
          courseContent: {
            html5: 'file.html',
            html5Embed: '<html></html>',
          },
        };

        // when
        const { result } = setup({ loading: false, course });
        await result.instance().handleTrainingClick('terminated')();

        // then
        expect(result.instance().canEvaluate()).toBeTruthy();
      });

      describe('when course was already evaluated before', () => {
        it('does not ask for a course evaluation', async () => {
          // given
          const courseRatedByYou = {
            id: 1,
            title: 'new course',
            description: 'new course description',
            durationInSeconds: 1234,
            stoppedAt: 12,
            views: 12,
            dateUpload: '2017-04-20T00:00:00.000Z',
            type: 'VIDEO',
            status: 'finished',
            isfavorite: true,
            relatedCourses: [],
            ratedByYou: 'true',
            courseContent: {
              html5: 'file.html',
              html5Embed: '<html></html>',
            },
          };

          // when
          const { result } = setup({ loading: false, course: courseRatedByYou });
          await result.instance().handleTrainingClick('terminated')();

          // then
          expect(result.instance().canEvaluate()).toBeFalsy();
        });
      });
    });

    describe('when starting/resuming a training course', () => {
      describe('when course type is web', () => {
        it('opens a new tab', async () => {
          // given
          const course = {
            id: 1,
            title: 'new course',
            description: 'new course description',
            durationInSeconds: 1234,
            stoppedAt: 12,
            views: 12,
            dateUpload: '2017-04-20T00:00:00.000Z',
            type: 'WEB',
            status: 'pending',
            isfavorite: true,
            relatedCourses: [],
            ratedByYou: 'false',
            courseContent: {
              web: 'google.com',
              video: null,
              html5: null,
            },
          };

          // when
          const { result } = setup({ loading: false, course });
          await result.instance().handleTrainingClick('initialized')();

          // then
          expect(global.open).toBeCalledWith(course.courseContent.web, '_blank');
        });
      });

      describe('when course type is video', () => {
        it('redirects to course details page', async () => {
          // given
          const course = {
            id: 1,
            title: 'new course',
            description: 'new course description',
            durationInSeconds: 1234,
            stoppedAt: 12,
            views: 12,
            dateUpload: '2017-04-20T00:00:00.000Z',
            type: 'VIDEO',
            status: 'pending',
            isfavorite: true,
            relatedCourses: [],
            ratedByYou: 'false',
            courseContent: {
              web: null,
              video: 'youtube.com',
              html5: null,
            },
          };

          // when
          const { result, props } = setup({ loading: false, course });
          await result.instance().handleTrainingClick('initialized')();

          // then
          expect(props.history.push).toBeCalledWith(`/training/courses/${course.id}/video`);
        });
      });
    });
  });
});
