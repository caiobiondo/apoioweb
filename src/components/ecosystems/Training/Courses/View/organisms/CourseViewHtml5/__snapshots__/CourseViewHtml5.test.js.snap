// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainingCourseViewHtml5 renders correctly when loading 1`] = `
<Loading
  background="transparent"
  visible={true}
  zIndex={9999}
/>
`;

exports[`TrainingCourseViewHtml5 renders correctly when the course is defined 1`] = `
<CourseViewHtml5styles__Main>
  <withRouter(CourseViewHeader) />
  <CourseViewHtml5styles__TrainingCourseThumbnailWrapper>
    <CourseViewHtml5styles__Html5Wrapper>
      <iframe
        allowFullScreen={true}
        frameBorder="0"
        src="about:blank"
        title="new course"
      />
    </CourseViewHtml5styles__Html5Wrapper>
  </CourseViewHtml5styles__TrainingCourseThumbnailWrapper>
  <CourseDescription
    course={
      Object {
        "courseContent": Object {
          "html5": "file.html",
          "html5Embed": "<html></html>",
        },
        "dateUpload": "2017-04-20T00:00:00.000Z",
        "description": "new course description",
        "durationInSeconds": 1234,
        "id": 1,
        "isfavorite": true,
        "ratedByYou": "false",
        "relatedCourses": Array [],
        "status": "started",
        "stoppedAt": 12,
        "title": "new course",
        "type": "VIDEO",
        "views": 12,
      }
    }
  />
  <CourseViewHtml5styles__TrainingCourseActions>
    <CourseViewHtml5styles__TrainingCourseActionButtonWrapper
      key="finish"
    >
      <FlatButton
        backgroundColor="#f8f8f8"
        disabled={false}
        fullWidth={false}
        hoverColor="#f8f8f8"
        icon={
          <Icon
            file="ico_play_circle"
          />
        }
        label="Finalizar treinamento"
        labelPosition="after"
        labelStyle={
          Object {
            "color": "#f3971f",
            "fontFamily": "Roboto-Medium",
            "paddingLeft": "0px",
            "paddingRight": "40px",
            "textTransform": "uppercase",
          }
        }
        onClick={[Function]}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        primary={false}
        secondary={false}
      />
    </CourseViewHtml5styles__TrainingCourseActionButtonWrapper>
    <CourseViewHtml5styles__TrainingCourseActionButtonWrapper
      key="list"
    >
      <FlatButton
        backgroundColor="#f8f8f8"
        disabled={false}
        fullWidth={false}
        hoverColor="#f8f8f8"
        icon={
          <Icon
            file="ico_plus"
          />
        }
        label="Minha Lista"
        labelPosition="after"
        labelStyle={
          Object {
            "color": "#f3971f",
            "fontFamily": "Roboto-Medium",
            "paddingLeft": "0px",
            "paddingRight": "40px",
            "textTransform": "uppercase",
          }
        }
        onClick={[Function]}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        primary={false}
        secondary={false}
      />
    </CourseViewHtml5styles__TrainingCourseActionButtonWrapper>
  </CourseViewHtml5styles__TrainingCourseActions>
  <CourseRating
    course={
      Object {
        "courseContent": Object {
          "html5": "file.html",
          "html5Embed": "<html></html>",
        },
        "dateUpload": "2017-04-20T00:00:00.000Z",
        "description": "new course description",
        "durationInSeconds": 1234,
        "id": 1,
        "isfavorite": true,
        "ratedByYou": "false",
        "relatedCourses": Array [],
        "status": "started",
        "stoppedAt": 12,
        "title": "new course",
        "type": "VIDEO",
        "views": 12,
      }
    }
  />
  <RelatedCourses
    courses={Array []}
  />
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label={
              <FormattedMessage
                  id="ok"
                  values={Object {}}
              />
          }
          labelPosition="after"
          labelStyle={
              Object {
                  "fontFamily": "Roboto-Regular",
                }
          }
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
      />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    key="feedbackModal"
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
    title=""
    titleStyle={
      Object {
        "fontFamily": "NaturaSans-Bold",
        "fontSize": "25px",
      }
    }
  />
</CourseViewHtml5styles__Main>
`;

exports[`TrainingCourseViewHtml5 renders correctly when the course is not defined 1`] = `""`;
